### DISCUSSION TITLE
Data Provider extensibility

### DISCUSSION THREAD

**Discussion by luismanez** [microsoft-search/pnp-modern-search#2054] [Q&A] [answered]
Just wondering if it´s able to extend the Search Results webpart with custom Data providers (SP, Graph + custom ones)? I don´t think so, but want to confirm, and also check if is in the roadmap.
I understand the "recommended" way to go, would be to create a Graph custom Connector, and use the pnp SRWP with Graph data provider. However Graph connectors are still very limited (based on user licenses, issues with external users...), so, would be good to be able to extend the webpart with custom Data Providers.

Many thanks!

**Comment by FranckyC** [comment, marked as answer]
Hi Luis, actually we don't plan to include any other sources in a near future BUT we could certainly extend the solution to be able to create custom ones as "plugins". It would be fairly easy to do I think as it has been designed this way.

**Reply by luismanez** [reply]
Thanks Franck. Just to double check then, with current code, we cannot add a custom Data Provider using the Extensibility library, right? In other words, when you say "we could extend the solution", you mean changing code in the core search library (search-parts folder). As far as I see in the code, I think I´m right, but just to confirm.
Thanks again.

**Reply by FranckyC** [reply]
You're right, with the current code we can't add a new data source. We would need to update the extensibility library + core code to support this use case the same way we do for layouts.

**Reply by edarroudi** [reply]
That would be very nice!
